### PR TITLE
Frontend refactor: scss

### PR DESCRIPTION
**Pull request by kristjanjansen**

### CODE REVIEW

File: resources/assets/sass/base/_base.colors.scss [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+// Utility colors
+
+$color-text-link: #36a1eb;
+
+$color-fill-success: $brand-green;
+$color-fill-success-dark: $brand-green-dark;
+
+// New colors
+
+$color-dark-gray: #111;
```
Comment (kristjanjansen) [inline resources/assets/sass/base/_base.colors.scss]: Just a comment: I pulled out some hardcoded gray values (mostly from mixins) which were coupled with alpha channels. At some point this needs simplification

File: resources/assets/sass/base/_base.layout.scss [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+$breakpoint-desktop: 1200px;
+$breakpoint-desktop-medium: 1440px;
+$breakpoint-desktop-large: 1520px;
+
+@include add-breakpoint('tablet-portrait', $breakpoint-tablet-portrait);
+@include add-breakpoint('tablet', $breakpoint-tablet);
+@include add-breakpoint('desktop', $breakpoint-desktop);
+@include add-breakpoint('desktop-medium', $breakpoint-desktop-medium);
+@include add-breakpoint('desktop-large', $breakpoint-desktop-large);
+@include add-breakpoint('vertical-small', (min-width $breakpoint-tablet) (max-height 640px));
```
Comment (kristjanjansen) [inline resources/assets/sass/base/_base.layout.scss]: Perhaps add a comment what are ```vertical-*``` breakpoints all about?

File: resources/assets/sass/base/_base.layout.scss [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+				padding: 0 span(2);
+			} @else {
+
+				padding: 0 12px;
+			}
+		}
+
+		@include respond-to(desktop) {
+
+			@include container(1152px);
```
Comment (kristjanjansen) [inline resources/assets/sass/base/_base.layout.scss]: Where from this px value is coming? can it be calculated using ```$breakpoint-desktop``` or smth?

File: resources/assets/sass/base/_base.layout.scss [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+				padding: 0 84px;
+			} @else {
+
+				padding: 0;
+			}
+		}
+
+		@include respond-to(desktop-medium) {
+
+			@include container(1248px);
```
Comment (kristjanjansen) [inline resources/assets/sass/base/_base.layout.scss]: ..same here

File: resources/assets/sass/base/_base.layout.scss [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+				padding: 0 90px;
+			} @else {
+
+				padding: 0;
+			}
+		}
+
+		@include respond-to(desktop-large) {
+
+			@include container(1344px);
```
Comment (kristjanjansen) [inline resources/assets/sass/base/_base.layout.scss]: ..same here

File: resources/assets/sass/base/_base.mixins.scss [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+		-moz-osx-font-smoothing: auto;
+	}
+}
+
+@mixin text-rendering($value: optimizeLegibility) {
+
+	text-rendering: $value; // Joseph Pearson also points out that on slower machines (such as mobile devices) it can negatively impact page load when applied to large blocks of text.
+}
+
+@mixin vertical-center {
```
Comment (kristjanjansen) [inline resources/assets/sass/base/_base.mixins.scss]: As we use flexbox everywhere, can we replace it with flexbox vertical centering?

File: resources/assets/sass/base/_base.mixins.scss [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+	&:after {
+
+		@if $direction == up { $translateAfter: -1px; } @else { $translateAfter: 1px; };
+
+		transform: translateX($translateAfter) rotate(-135deg);
+	}
+}
+
+@mixin triangle($direction, $size: 6px, $color: $color-dark-gray3) {
```
Comment (kristjanjansen) [inline resources/assets/sass/base/_base.mixins.scss]: this looks really bizy :)

File: resources/assets/sass/base/_base.scss [hunk 8]
```diff
@@ -0,0 +1,92 @@
+$path-images: '../images/' !default;
```
Comment (kristjanjansen) [inline resources/assets/sass/base/_base.scss]: Can we remove these variables and commented-out @imports?

File: resources/assets/sass/components/_destination-info.scss [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+		font-weight: $font-weight-medium;
+		line-height: 1.5;
+
+		a {
+
+			font-size: $font-size-normal;
+			font-weight: $font-weight-bold;
+			text-decoration: none;
+
+			.c-destination-info.m-red & {color: $brand-red-dark;}
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_destination-info.scss]: Can we use mixins with color map for such modifier injects?

File: resources/assets/sass/base/_base.colors.scss (hunk 1 above)
Comment (apani) [inline resources/assets/sass/base/_base.colors.scss]: I'll do the grouping and renaming of the colors

File: resources/assets/sass/components/_destination-info.scss [hunk 10]
```diff
@@ -0,0 +1,54 @@
+.c-destination-info {}
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_destination-info.scss]: for file name, lets use ```_destination_info.scss``` or ```destination/_info.scss```

File: resources/assets/sass/components/_button.scss [hunk 11]
```diff
@@ -0,0 +1,51 @@
+// Button
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_button.scss]: Move to Blade if it is not done yet

File: resources/assets/sass/components/_destination-info.scss [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+	}
+
+		.c-destination-info__definition-list-term {
+
+			float: left;
+			clear: left;
+			min-width: 30%;
+			margin-bottom: 18px;
+
+			.c-destination-info.m-red & {color: $brand-red-dark;}
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_destination-info.scss]: ...same here

File: resources/assets/sass/components/_destination.scss [hunk 13]
```diff
@@ -0,0 +1,58 @@
+// Destination
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_destination.scss]: Move comment to Blade if it is not done yet

File: resources/assets/sass/base/_base.layout.scss (hunk 2 above)
Comment (apani) [inline resources/assets/sass/base/_base.layout.scss]: Ok

File: resources/assets/sass/components/_footer.scss [hunk 14]
```diff
@@ -0,0 +1,132 @@
+.c-footer {
+
+	padding: 132px 0 48px;
+	background: url('#{$path-images}/footer-bg.jpg') no-repeat top center;
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_footer.scss]: remove

File: resources/assets/sass/components/_forum.scss [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+
+				&:nth-child(even) {
+
+					background: none;
+					box-shadow: none;
+				}
+			}
+		}
+
+			.c-forum__list-item-profile {
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_forum.scss]: According to BEM, should it not be something like

    .c-forum__list-item--profile ?

File: resources/assets/sass/base/_base.layout.scss (hunk 3 above)
Comment (apani) [inline resources/assets/sass/base/_base.layout.scss]: Base padding, can be anything - currently just a dummy value, will be changed in the future (based on design)

File: resources/assets/sass/components/_grid.scss [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+	flex-wrap: wrap;
+}
+
+	.c-columns__item {
+
+		max-width: 100%;
+
+		.c-columns.m-3-cols & {
+
+			flex: 1 0 320px;
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_grid.scss]: Should 320px be variable / calculated somehow?

File: resources/assets/sass/components/_gallery.scss [hunk 17]
```diff
@@ -0,0 +1,109 @@
+// Gallery
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_gallery.scss]: Move comment to Blade if it is not done yet

File: resources/assets/sass/components/_list.scss [hunk 18]
```diff
@@ -0,0 +1,164 @@
+// Lists
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_list.scss]: Move comment to Blade if it is not done yet

File: resources/assets/sass/base/_base.layout.scss (hunk 4 above)
Comment (apani) [inline resources/assets/sass/base/_base.layout.scss]: 1152px container / 28 cols *2 ~ 84px (narrow container is 4 columns smaller) - span(2) did not work in this

File: resources/assets/sass/components/_list.scss [hunk 19]
```diff
@@ -0,0 +1,164 @@
...
+				left: 0;
+				top: 3px;
+				width: 7px;
+				height: 7px;
+				border-radius: 50%;
+			}
+		}
+	}
+
+	.c-list.m-dot.m-yellow .c-list__item:before {background-color: $brand-yellow;}
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_list.scss]: Can we use mixin here?

File: resources/assets/sass/components/_list.scss [hunk 20]
```diff
@@ -0,0 +1,164 @@
...
+			display: block;
+			width: 48px;
+			height: 48px;
+
+			svg {
+
+				height: 100%;
+				width: 100%;
+
+				.c-list__item.m-red & {fill: $brand-red;}
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_list.scss]: ...same here

File: resources/assets/sass/base/_base.layout.scss (hunk 5 above)
Comment (apani) [inline resources/assets/sass/base/_base.layout.scss]: 1248px container / 28 cols *2 ~ 90px (narrow container is 4 columns smaller) - span(2) did not work in this

File: resources/assets/sass/base/_base.mixins.scss (hunk 6 above)
Comment (apani) [inline resources/assets/sass/base/_base.mixins.scss]: yes we can

File: resources/assets/sass/base/_base.mixins.scss (hunk 7 above)
Comment (apani) [inline resources/assets/sass/base/_base.mixins.scss]: remove?

File: resources/assets/sass/components/_logo.scss [hunk 21]
```diff
@@ -0,0 +1,16 @@
+.c-logo {
+
+	display: block;
+	height: 132px;
+	width: 279px;
+	background: url('#{$path-svg}tripee-logo.svg') no-repeat center center;
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_logo.scss]: Remove url part, lets move the setting the logo background image to Blade side (dynamic logos etc).

File: resources/assets/sass/components/_logo.scss [hunk 22]
```diff
@@ -0,0 +1,16 @@
...
+
+	&.m-small {
+
+		width: 202px;
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_logo.scss]: use variables

File: resources/assets/sass/components/_masthead.scss [hunk 23]
```diff
@@ -0,0 +1,160 @@
+// Masthead
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_masthead.scss]: Move comment to Blade if it is not done yet

File: resources/assets/sass/components/_masthead.scss [hunk 24]
```diff
@@ -0,0 +1,160 @@
...
+// default   - Destination masthead.
+//
+// Styleguide 8.1
+
+.c-masthead {
+
+	position: relative;
+	width: 100%;
+	height: 240px;
+	//background-image: url('#{$path-images}/cover-bg-default.jpg');
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_masthead.scss]: remove

File: resources/assets/sass/base/_base.scss (hunk 8 above)
Comment (apani) [inline resources/assets/sass/base/_base.scss]: If needed I can remove those path variables ... , will remove @imports

File: resources/assets/sass/base/_base.mixins.scss (hunk 7 above)
Comment (kristjanjansen) [inline resources/assets/sass/base/_base.mixins.scss]: see https://github.com/tripikad/trip2/pull/336/files#r42606231

File: resources/assets/sass/components/_footer.scss [hunk 25]
```diff
@@ -0,0 +1,132 @@
...
+
+	&:after {
+
+		content: '';
+		position: absolute;
+		top: 0;
+		left: 0;
+		width: 100%;
+		height: 100%;
+		background: rgba($color-dark-gray, 0.6);
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_footer.scss]: see https://github.com/tripikad/trip2/pull/336/files#r42606231

File: resources/assets/sass/components/_masthead.scss [hunk 26]
```diff
@@ -0,0 +1,160 @@
...
+
+	&:after {
+
+		content: '';
+		position: absolute;
+		top: 0;
+		left: 0;
+		width: 100%;
+		height: 100%;
+		background: rgba($color-dark-gray2, 0.4);
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_masthead.scss]: See https://github.com/tripikad/trip2/pull/336/files#r42606231

File: resources/assets/sass/components/_news.scss [hunk 27]
```diff
@@ -0,0 +1,110 @@
+// News
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_news.scss]: Move comment to Blade if it is not done yet

File: resources/assets/sass/components/_offer.scss [hunk 28]
```diff
@@ -0,0 +1,78 @@
+// Offer
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_offer.scss]: Move comment to Blade if it is not done yet

File: resources/assets/sass/components/_destination-info.scss (hunk 9 above)
Comment (apani) [inline resources/assets/sass/components/_destination-info.scss]: Probably

File: resources/assets/sass/components/_profile.scss [hunk 29]
```diff
@@ -0,0 +1,151 @@
+// Profile
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_profile.scss]: Move comment to Blade if it is not done yet

File: resources/assets/sass/components/_search.scss [hunk 30]
```diff
@@ -0,0 +1,407 @@
...
+		        	height: 24px;
+		        	width: 24px;
+		        }
+			}
+
+			&.m-destinations {
+
+				&:before {
+
+					background: url('#{$path-images}/icon-offer.svg');
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_search.scss]: always being advocating keeping SVGs in Blade side. Not sure about these. I'd like concistency but...

File: resources/assets/sass/components/_search.scss [hunk 31]
```diff
@@ -0,0 +1,407 @@
...
+			left: 0;
+			display: block;
+			width: 100%;
+			height: 54px;
+			background-image: linear-gradient(to bottom, rgba($color-white, 0), rgba($color-white, 1));
+			pointer-events: none;
+		}
+	}
+
+
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_search.scss]: https://38.media.tumblr.com/tumblr_m55pfm54011r13ra2.gif

File: resources/assets/sass/components/_tags.scss [hunk 32]
```diff
@@ -0,0 +1,49 @@
+// Tags
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_tags.scss]: Move comment to Blade if it is not done yet

File: resources/assets/sass/components/_title.scss [hunk 33]
```diff
@@ -0,0 +1,31 @@
+// Title
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_title.scss]: Move comment to Blade if it is not done yet

File: resources/assets/sass/components/_travel-letter.scss [hunk 34]
```diff
@@ -0,0 +1,34 @@
+.c-travel-letter {}
```
Comment (kristjanjansen) [inline resources/assets/sass/components/_travel-letter.scss]: What is travel-letter? Blog? Lets keep it consistent and rename blog to something better in one go.

File: resources/assets/sass/regions/_destination.scss [hunk 35]
```diff
@@ -0,0 +1,295 @@
...
+					padding-bottom: 48px;
+
+					@include respond-to(tablet) {
+
+						@include span(8 of 28);
+
+						padding-bottom: 0;
+						margin-top: -48px;
+
+						.r-destination__about.m-red & {box-shadow: 0 12px 36px rgba($brand-red-dark, 1);}
```
Comment (kristjanjansen) [inline resources/assets/sass/regions/_destination.scss]: Mixin?

File: resources/assets/sass/regions/_home.scss [hunk 36]
```diff
@@ -0,0 +1,285 @@
...
+
+            @include wrap('narrow');
+        }
+
+            .r-home__destinations-action {
+
+                padding: 24px 0;
+            }
+
+    // Ablot
```
Comment (kristjanjansen) [inline resources/assets/sass/regions/_home.scss]: https://www.youtube.com/watch?v=R2nkvtKvIoQ

**Pull request closed without merging**